tools/firtool/firtool.cpp

```diff
@@ -112,7 +118,7 @@ processBuffer(std::unique_ptr<llvm::MemoryBuffer> ownedBuffer,
   if (inputFormat == InputFIRFile) {
     firrtl::FIRParserOptions options;
     options.ignoreInfoLocators = ignoreFIRLocations;
-    module = importFIRRTL(sourceMgr, &context, options);
+    module = importFIRRTL(sourceMgr, annotations, &context, options);
```

Comment:
The top level dance here is a bit weird to me because both the caller and callee of importFIRRTL has to know about JSON stuff.  I'd recommend that firtool.cpp pass in annotations as a filename, or StringRef for the buffer, and put all the JSON parsing stuff (JSON.h etc) into the caller.

Per the macro comment, if might make sense to make annotation parsing be a separate call from importFIRRTL.

Reply:
I think this is all cleaned up. Now `firtool` no longer has to know about JSON. The actual annotation file is pulled in by the source manager in the parser.